### ISSUE TITLE
Python extension is required when using VS Code notebooks with IJava kernel

### ISSUE THREAD

**Issue by ronglums** [microsoft/vscode-jupyter#4526]
When using VS Code notebooks with the IJava kernel, the Java kernel doesn't show up in the kernel picker unless the Python extension is installed. The expectation is the Java scenario doesn't have any dependencies on Python (and the Python extension).

![image](https://user-images.githubusercontent.com/26419852/106036639-733c0a00-608a-11eb-9770-bb39fa540411.png)

**Comment by DonJayamanne** [comment]
@IanMatthewHuff has some background info about this issue.

**Comment by greazer** [comment]
Should be fixed by #4965

**Comment by DavidKutu** [comment]
I couldn't repro this:
![image](https://user-images.githubusercontent.com/20347396/111706037-dc283e80-87fe-11eb-875c-b67ec0f11aaf.png)
We might've fixed it already

**Comment by DavidKutu** [comment]
I'm moving this to validate, it should be fixed on our next release.

If somebody finds a repro, feel free to put it back.

**Comment by IanMatthewHuff** [comment]
Adding David just to distinguish the work from the validation.

**Comment by IanMatthewHuff** [comment]
IJava install options, as this might make a difference in where the kernelspec is placed.

```
Install the java kernel.

optional arguments:
  -h, --help            show this help message and exit
  --user                Install to the per-user kernel registry.
  --sys-prefix          Install to Python's sys.prefix. Useful in conda/virtual environments.
  --prefix PREFIX       Specify a prefix to install to, e.g. an env. The kernelspec will be installed in PREFIX/share/jupyter/kernels/
  --replace             Replace any existing kernel spec with this name.
  --classpath ENV       A file path separator delimited list of classpath entries that should be available to the user code. **Important:** no matter what OS, this should use forward slash
                        "/" as the file separator. Also each path may actually be a simple glob.
  --comp-opts ENV       A space delimited list of command line options that would be passed to the `javac` command when compiling a project. For example `-parameters` to enable retaining
                        parameter names for reflection.
  --startup-scripts-path ENV
                        A file path seperator delimited list of `.jshell` scripts to run on startup. This includes ijava-jshell-init.jshell and ijava-display-init.jshell. **Important:** no
                        matter what OS, this should use forward slash "/" as the file separator. Also each path may actually be a simple glob.
  --startup-script ENV  A block of java code to run when the kernel starts up. This may be something like `import my.utils;` to setup some default imports or even `void sleep(long time) {
                        try {Thread.sleep(time); } catch (InterruptedException e) { throw new RuntimeException(e); }}` to declare a default utility method to use in the notebook.
  --timeout ENV         A duration specifying a timeout (in milliseconds by default) for a _single top level statement_. If less than `1` then there is no timeout. If desired a time may be
                        specified with a `TimeUnit` may be given following the duration number (ex `"30 SECONDS"`).
```

**Comment by IanMatthewHuff** [comment]
```
(javaEnv) MININT-ST2TT53:~ ianhuff$ python ./Desktop/ijava-1.3.0/install.py --sys-prefix
./Desktop/ijava-1.3.0/install.py:164: DeprecationWarning: replace is ignored. Installing a kernelspec always replaces an existing installation
  install_dest = KernelSpecManager().install_kernel_spec(
Installed java kernel into "/Users/ianhuff/opt/miniconda3/envs/javaEnv/share/jupyter/kernels/java"
```

**Comment by IanMatthewHuff** [comment]
```
(javaEnv) MININT-ST2TT53:~ ianhuff$ python ./Desktop/ijava-1.3.0/install.py --user
./Desktop/ijava-1.3.0/install.py:164: DeprecationWarning: replace is ignored. Installing a kernelspec always replaces an existing installation
  install_dest = KernelSpecManager().install_kernel_spec(
Installed java kernel into "/Users/ianhuff/Library/Jupyter/kernels/java"
```

**Comment by IanMatthewHuff** [comment]
@DavidKutu @ronglums  Looks like this was the issue. On the IJava page they use install.py --sys-prefix as their example install command, which I used. This places the kernelspec in a env local location, so without the python interpreter to locate the env the kernel spec can't be found (the situation that I was in the first time I tried this). However with the --user option the kernelspec was installed into the user specific kernelspec location and I was able to run the kernel without the python extension installed / activated.
